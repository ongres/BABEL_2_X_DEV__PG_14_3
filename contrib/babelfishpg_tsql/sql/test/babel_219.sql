-- This test should be run without installing the babelfishpg_tsql extension
-- BABEL-219 test a domain named varchar in schema other than sys
-- is not affected by the fix of BABEL-219
create domain public.varchar as pg_catalog.varchar(2) check (char_length(value) < 1);
select cast('a' as public.varchar); -- throw error
select cast('' as public.varchar);
select cast('a' as varchar); -- pg_catalog.varchar should work

show search_path;
-- Explicitly add pg_catalog to tail of search_path,
-- to force varchar default to public.varchar
select set_config('search_path', current_setting('search_path') || ', pg_catalog', false);
-- Set tsql dialet so the fix for BABEL-219 can kick in
SET babelfishpg_tsql.sql_dialect = 'tsql';
select cast('a' as varchar); -- varchar default to public.varchar. should fail exactly the same way as explicitly specifying public.varchar
select cast('' as varchar); -- varchar default to public.varchar. should pass
create table t1(col varchar);
insert into t1 (col) select 'a'; -- fail
insert into t1 (col) select ''; -- pass
select * from t1;
-- verify behavior of public.varchar is unchanged in tsql dialect
select cast('a' as public.varchar); -- fail
select cast('' as public.varchar); -- pass
create table t2(col public.varchar);
insert into t1 (col) select 'a'; -- fail
insert into t1 (col) select ''; -- pass
select * from t1;

-- Clean up
drop table t1;
drop table t2;
set babelfishpg_tsql.sql_dialect = 'postgres';
-- Reset search_path
set search_path to "$user", public;
show search_path;
