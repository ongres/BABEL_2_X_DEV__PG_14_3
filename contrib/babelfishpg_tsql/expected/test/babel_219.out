-- This test should be run without installing the babelfishpg_tsql extension
-- BABEL-219 test a domain named varchar in schema other than sys
-- is not affected by the fix of BABEL-219
create domain public.varchar as pg_catalog.varchar(2) check (char_length(value) < 1);
select cast('a' as public.varchar); -- throw error
ERROR:  value for domain public."varchar" violates check constraint "varchar_check"
select cast('' as public.varchar);
 varchar 
---------
 
(1 row)

select cast('a' as varchar); -- pg_catalog.varchar should work
 varchar 
---------
 a
(1 row)

show search_path;
   search_path   
-----------------
 "$user", public
(1 row)

-- Explicitly add pg_catalog to tail of search_path,
-- to force varchar default to public.varchar
select set_config('search_path', current_setting('search_path') || ', pg_catalog', false);
         set_config          
-----------------------------
 "$user", public, pg_catalog
(1 row)

-- Set tsql dialet so the fix for BABEL-219 can kick in
SET babelfishpg_tsql.sql_dialect = 'tsql';
select cast('a' as varchar); -- varchar default to public.varchar. should fail exactly the same way as explicitly specifying public.varchar
ERROR:  value for domain "varchar" violates check constraint "varchar_check"
select cast('' as varchar); -- varchar default to public.varchar. should pass
 varchar 
---------
 
(1 row)

create table t1(col varchar);
insert into t1 (col) select 'a'; -- fail
ERROR:  value for domain "varchar" violates check constraint "varchar_check"
insert into t1 (col) select ''; -- pass
select * from t1;
 col 
-----
 
(1 row)

-- verify behavior of public.varchar is unchanged in tsql dialect
select cast('a' as public.varchar); -- fail
ERROR:  value for domain "varchar" violates check constraint "varchar_check"
select cast('' as public.varchar); -- pass
 varchar 
---------
 
(1 row)

create table t2(col public.varchar);
insert into t1 (col) select 'a'; -- fail
ERROR:  value for domain "varchar" violates check constraint "varchar_check"
insert into t1 (col) select ''; -- pass
select * from t1;
 col 
-----
 
 
(2 rows)

-- Clean up
drop table t1;
drop table t2;
set babelfishpg_tsql.sql_dialect = 'postgres';
-- Reset search_path
set search_path to "$user", public;
show search_path;
   search_path   
-----------------
 "$user", public
(1 row)

